data_augument_yolo/data_augumentation.py

```diff
@@ -25,34 +61,122 @@ def add_noise(self, noise_type='gaussian', mean=0, var=10, salt_pepper_ratio=0.0
+    def adjust_brightness(self, scale: float = 1.0) -> None:
+        """
+        Adjusts the brightness of the image.
+
+        Args:
+            scale (float): Scaling factor for brightness. Values greater than 1 increase brightness.
+        """
         hsv_image = cv2.cvtColor(self.modified_image, cv2.COLOR_BGR2HSV)
         hsv_image[..., 2] = np.clip(hsv_image[..., 2] * scale, 0, 255)
         self.modified_image = cv2.cvtColor(hsv_image, cv2.COLOR_HSV2BGR)
 
-    def perspective_transform(self, max_angle=10):
+    def perspective_transform(self, max_angle: float = 10) -> None:
+        """
+        Applies a perspective transformation to the image.
+
+        Args:
+            max_angle (float): Maximum angle of transformation.
+        """
         h, w = self.modified_image.shape[:2]
         pts1 = np.float32([[0, 0], [w, 0], [0, h], [w, h]])
         dx, dy = np.random.uniform(-max_angle, max_angle, size=4).reshape(2, 2)
         pts2 = pts1 + np.float32([[dx[0], dy[0]], [dx[1], dy[0]], [dx[0], dy[1]], [dx[1], dy[1]]])
         matrix = cv2.getPerspectiveTransform(pts1, pts2)
         self.modified_image = cv2.warpPerspective(self.modified_image, matrix, (w, h))
 
-    def apply_transformations(self, noise_type=None, rotation_angle=0, brightness_scale=1.0, perspective_angle=0):
+    def apply_scaling(self, scale_factor: float) -> None:
+        """
+        Scales the image by a given factor.
+
+        Args:
+            scale_factor (float): The factor by which to scale the image.
+        """
+        h, w = self.modified_image.shape[:2]
+        self.modified_image = cv2.resize(self.modified_image, (int(w * scale_factor), int(h * scale_factor)))
+
+    def flip_image(self, direction: str = 'horizontal') -> None:
+        """
+        Flips the image horizontally or vertically.
+
+        Args:
+            direction (str): 'horizontal' or 'vertical'.
+        """
+        if direction == 'horizontal':
+            self.modified_image = cv2.flip(self.modified_image, 1)
+        elif direction == 'vertical':
+            self.modified_image = cv2.flip(self.modified_image, 0)
+
+    def crop_image(self, x: int, y: int, width: int, height: int) -> None:
+        """
+        Crops the image to a specified rectangle.
+
+        Args:
+            x (int): x-coordinate of the top-left corner.
+            y (int): y-coordinate of the top-left corner.
+            width (int): Width of the rectangle to crop.
+            height (int): Height of the rectangle to crop.
+        """
+        self.modified_image = self.modified_image[y:y + height, x:x + width]
+    
+    def adjust_contrast(self, alpha: float) -> None:
+        """
+        Adjusts the contrast of the image.
+
+        Args:
+            alpha (float): Contrast control (1.0 = no change, <1.0 = lower contrast, >1.0 = higher contrast).
+        """
+        self.modified_image = cv2.convertScaleAbs(self.modified_image, alpha=alpha, beta=0)
```

Comment:
Anh thấy nên để input là ảnh raw output là ảnh modified. Em sửa lại toàn bộ các hàm nhé